spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/SelectBuilder.java

```diff
@@ -486,6 +486,16 @@ interface SelectJoin extends SelectLock, BuildSelect {
 		 * @see SQL#table(String)
 		 */
 		SelectOn leftOuterJoin(TableLike table);
+
+		/**
+		 * Declare a {@code FULL OUTER JOIN} {@link Table}.
+		 *
+		 * @param table must not be {@literal null}.
+		 * @return {@code this} builder.
+		 * @see Join
+		 * @see SQL#table(String)
+		 */
+		SelectOn fullOuterJoin(TableLike table);
```

Comment:
I wonder whether we should introduce a `JoinType` type to be more flexible and not always require a new method per join type. That type could describe `INNER JOIN`, `LEFT/RIGHT` joins, outer, cross, natural and many more of those.

Reply:
I agree, I will replace `fullOuterJoin` with `join` with an additional parameter.